decode.go

```diff
@@ -345,16 +350,21 @@ var ufixed64type = reflect.TypeOf(Ufixed64(0))
 // Handle decoding of slices
 func (de *decoder) slice(slval reflect.Value, vb []byte) error {
 
+
 	// Find the element type, and create a temporary instance of it.
 	eltype := slval.Type().Elem()
 	val := reflect.New(eltype).Elem()
 
 	// Decide on the wiretype to use for decoding.
 	var wiretype int
 	switch eltype.Kind() {
-	case reflect.Bool, reflect.Int32, reflect.Int64,
-		reflect.Uint32, reflect.Uint64:
+	case reflect.Bool, reflect.Int32, reflect.Int64, reflect.Int,
+		reflect.Uint32, reflect.Uint64, reflect.Uint:
+		if (eltype.Kind() == reflect.Int || eltype.Kind() == reflect.Uint) && eltype.Size() < 8 {
+			return errors.New("detected a 32bit machine, please either use (u)int64 or (u)int32")
```

Comment:
I'm wondering if a panic is not better in this case since it's highly likely that the person who decodes with now an invalid struct will do it again, so in my opinion he should be forced to change his code, in the spirit of "detect the error as quickly as possible".

Reply:
Good question. I tend to stay on the `error` side here, even though the user most probably cannot continue with that error. But it looks like this could be abused to create a `panic` from an attacker - so I prefer to keep `error` here.